modules/exploits/linux/local/runc_cwd_priv_esc.rb

```diff
@@ -0,0 +1,143 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = NormalRanking # https://docs.metasploit.com/docs/using-metasploit/intermediate/exploit-ranking.html
+
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'runc (docker) File Descriptor Leak Privilege Escalation',
+        'Description' => %q{
+          All versions of runc <=1.1.11, as used by containerization technologies such as Docker engine,
+          and Kubernetes are vulnerable to an arbitrary file write.
+          Due to a file descriptor leak it is possible to mount the the host file system
+          with the permissions of runc (typically root).
+
+          Successfully tested on Ubuntu 22.04 with runc 1.1.7-0ubuntu1~22.04.1 using Docker build.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Rory McNamara' # Discovery
+        ],
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' => [[ 'Auto', {} ]],
+        'Privileged' => true,
+        'References' => [
+          [ 'URL', 'https://snyk.io/blog/cve-2024-21626-runc-process-cwd-container-breakout/'],
+          [ 'URL', 'https://github.com/opencontainers/runc/security/advisories/GHSA-xr7r-f8xq-vfvv'],
+          [ 'CVE', '2024-21626']
+        ],
+        'DisclosureDate' => '2024-01-31',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'AKA' => ['Leaky Vessels'],
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK]
+        },
+        'DefaultOptions' => {
+          'EXITFUNC' => 'thread',
+          'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+        }
+      )
+    )
+    register_advanced_options [
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ]),
```

Comment:
```suggestion
      OptString.new('WritableDir', [ true, 'A directory where we can write and execute files', '/tmp' ]),
```

Sometimes `/tmp` is mounted as `noexec`.

Reply:
good catch, that's from the example module. should prob be fixed there